persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/LogAppender.java

```diff
@@ -1,124 +1,207 @@
+            return (Boolean) deleteRequest.getIndexRequest().getResult();
+         } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+         }
+      });
    }
 
    @Override
-   public void run() {
+   public void accept(LogRequest request) {
       try {
-         FileProvider.Log logFile = fileProvider.getFileForLog();
-         int currentOffset = 0;
-         while (true) {
-            LogRequest request = queue.pop();
-            if (request != null) {
-               if (request.isClear()) {
-                  logFile.close();
-                  queue.notifyNoWait();
-                  request.pause();
-                  currentOffset = 0;
-                  logFile = fileProvider.getFileForLog();
-                  log.debug("Appending records to " + logFile.fileId);
-                  continue;
-               } else if (request.isStop()) {
-                  queue.notifyNoWait();
-                  break;
-               } else if (request.isPause()) {
-                  queue.notifyNoWait();
-                  request.pause();
-                  continue;
-               }
-               if (currentOffset + request.length() > maxFileSize) {
-                  // switch to next file
-                  logFile.close();
-                  compactor.completeFile(logFile.fileId);
-                  currentOffset = 0;
-                  logFile = fileProvider.getFileForLog();
-                  log.debug("Appending records to " + logFile.fileId);
-               }
-               long seqId = nextSeqId();
-               EntryRecord.writeEntry(logFile.fileChannel, request.getSerializedKey(), request.getSerializedMetadata(),
-                     request.getSerializedInternalMetadata(),
-                     request.getSerializedValue(), seqId, request.getExpiration(), request.getCreated(), request.getLastUsed());
-               int offset = request.getSerializedValue() == null ? ~currentOffset : currentOffset;
-               temporaryTable.set(request.getKey(), logFile.fileId, offset);
-               IndexRequest indexRequest = IndexRequest.update(request.getKey(), raw(request.getSerializedKey()),
-                     logFile.fileId, offset, request.length());
-               request.setIndexRequest(indexRequest);
-               indexQueue.put(indexRequest);
-               currentOffset += request.length();
-            } else {
-               if (syncWrites) {
-                  logFile.fileChannel.force(false);
-               }
-               queue.notifyAndWait();
-            }
+         if (logFile == null) {
+            logFile = fileProvider.getFileForLog();
+            log.debug("Appending records to " + logFile.fileId);
+         }
+         if (request.isClear()) {
+            logFile.close();
+            currentOffset = 0;
+            logFile = null;
+            // TODO: Technically other writes won't be notified if sync write is enabled until
+            // the timeout above
```

Comment:
Is there anything we can do this? Or do we need to remove the TODO and just comment the behaviour?

Reply:
I can't think of a better way to fix it. I can just comment that is the behavior. It is very similar to the behavior before.